### ISSUE TITLE
UsePam yes is set in new clusters with 4.7.0

### ISSUE THREAD

**Issue by bo0ts** [CrunchyData/postgres-operator#2538]
## Overview

We install and update the Crunchy PostgresOperator via OperatorHub in OpenShift 4.5. We recently updated from versin 4.6.2 to version 4.7.0.

After the update we deleted the pgo-config configmap and it was recreated by the operator. The configmap was not modified afterwards.

After the update creating a cluster fails during the execution of the `create-stanza` job with the error message:

```
time="2021-07-13T06:22:49Z" level=info msg="command to execute is [pgbackrest stanza-create  --db-host=10.134.9.102 --db-path=/pgdata/testcluster9]"
time="2021-07-13T06:22:49Z" level=info msg="output=[]"
time="2021-07-13T06:22:49Z" level=info msg="stderr=[WARN: unable to check pg-1: [UnknownError] remote-0 process on '10.134.9.102' terminated unexpectedly [255]\nERROR: [056]: unable to find primary cluster - cannot proceed\n]"
time="2021-07-13T06:22:49Z" level=fatal msg="command terminated with exit code 56"
```

## Environment

Please provide the following details:

- Platform: OpenShift 4.5
- Platform Version: 4.7.0
- PGO Image Tag: crunchy-postgres-ha:centos8-13.3-4.7.0
- Postgres Version 13

### REPRO

1. `pgo create cluster testcluster9 -n testnamespace`
2. Observe the containers spawned by the create-stanza job.

### EXPECTED

1. A cluster is created.

### ACTUAL

1. Describe what actually happens

## Logs

create-stanza:

```
time="2021-07-13T06:22:49Z" level=info msg="command to execute is [pgbackrest stanza-create  --db-host=10.134.9.102 --db-path=/pgdata/testcluster9]"
time="2021-07-13T06:22:49Z" level=info msg="output=[]"
time="2021-07-13T06:22:49Z" level=info msg="stderr=[WARN: unable to check pg-1: [UnknownError] remote-0 process on '10.134.9.102' terminated unexpectedly [255]\nERROR: [056]: unable to find primary cluster - cannot proceed\n]"
time="2021-07-13T06:22:49Z" level=fatal msg="command terminated with exit code 56"
```

pgbackrest-shared-repo:

```
NWRAP_ERROR(1) - nwrap_files_cache_reload: Unable to open '/tmp/nss_wrapper/pgbackrest-repo/passwd' readonly -1:No such file or directory
NWRAP_ERROR(1) - nwrap_files_getpwuid: Error loading passwd file
nss_wrapper: user exists
nss_wrapper: group exists
nss_wrapper: environment configured
Starting the pgBackRest repo
nss_wrapper: ssh configured
creating  /backrestrepo/testcluster9-backrest-shared-repo
The pgBackRest repo has been started
```

## Additional Information

A similar issue was fixed by setting `UsePam no` in the `clustername-backrest-repo-config`. When I checked the automatically created secret `testcluster9-backrest-repo-config` it contained `UsePam yes`.

When we create the secret ourselves before creating the cluster with `pgo create` creation succeeds. This seems like a regression to 4.6.2 where `UsePam no` was correctly set in the automatically created config.

**Comment by bo0ts** [comment]
I've fixed this problem by removing the secret `pgo-backrest-repo-config` from the operator namespace. 

On restart the operator recreated it with the correct configuration for sshd_config. That step seems to be missing from the "Upgrade" documentation and should be added. In general this is rather inconvenient, because that secret also contains our S3 credentials for backups and needs to be customized after every update.

**Comment by jkatz** [comment]
Did you run `pgo upgrade` or follow the manual instructions?

You should not need to clear out the Secret.

**Comment by bo0ts** [comment]
@jkatz Yes, like I said under "Repo" we used `pgo create cluster testcluster9 -n testnamespace`.

**Comment by jkatz** [comment]
@bo0ts That was not my question. I asked if you ran `pgo upgrade` or perform a manual update on the cluster. The reproduction steps you provided did not indicate that.

The reason I'm asking is to see if we do actually need to modify the manual guidance or if there was an issue with `pgo upgrade` itself.

**Comment by bo0ts** [comment]
@jkatz We ran `pgo upgrade` on the cluster.

**Comment by sunofsparda** [comment]
@bo0ts had the same issue in 4.7.0, perhaps because of https://github.com/CrunchyData/postgres-operator/issues/2478
We couldn't fix it, so performed rollback to 4.6.2 with ansible..

**Comment by andrewlecuyer** [comment]
Per the discussion above, it looks like it was possible to workaround this issue in PGO v4.

Furthermore, please note as of PGO v5.1, SSHD has been removed from the PGO architecture, and has been replaced with TLS.  Per the [PGO v5.1.0 relase notes](https://access.crunchydata.com/documentation/postgres-operator/latest/releases/5.1.0/):

> Previous versions of PGO relied on the use of ssh to take backups and store archive files on Kubernetes-managed storage. PGO v5.1 now uses mTLS to securely transfer and manage these files.

We therefore recommend [upgrading](https://access.crunchydata.com/documentation/postgres-operator/latest/upgrade/v4tov5/) to the latest version of PGO v5 (where the setting referenced in this issue is no longer applicable), and if you still have trouble with stanza creation, please let us know.

The docs for the latest version of PGO v5 can be found here:

https://access.crunchydata.com/documentation/postgres-operator/latest/
